packages/block-library/src/image/edit.js

```diff
@@ -171,6 +171,7 @@ export function ImageEdit( {
 	function onUploadError( message ) {
 		noticeOperations.removeAllNotices();
 		noticeOperations.createErrorNotice( message );
+		setTemporaryURL( undefined );
```

Comment:
I realized that we only need to reset the state. The `onSelectImage` will remove attributes for us. It's called during the error but with an undefined `media` argument.

Reply:
This is a weird one 🤔  - for me this change causes the php.ini upload limit error to be ignored, but only in the gallery:
![image-gallery-error](https://user-images.githubusercontent.com/3629020/156943445-48cf80fc-4216-4061-9a68-789505daa9b6.gif)

Seems like there are a couple of `gotchas` with how gallery versus individual images blocks are behaving with these errors, which would be good to work out why and resolve, or at least document. I will try and get a closer look at it later today.

Reply:
I had another look at it, and for some reason it seems like the standalone Image block goes through an extra render cycle which causes image to be removed and error to display, but this doesn't happen if the Image block is in a Gallery, but I can't work out why as the Gallery essentially just delegates the upload process to the Image block. I will take another look at it tomorrow if you don't have any ideas. 

We could just put the [setAttributes](https://github.com/WordPress/gutenberg/pull/39178/commits/de17283b98005198a3337423196f133ed4237b25) back, but it would be nice to know what this is happening.

Reply:
Thanks for testing, @glendaviesnz. I will revert `setAttributes` removal.